### PR TITLE
Feature/response activity form

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes
- Add response activity form

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/ThreeWActivityForm/schema.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    client_id: string;
+};
+type ActivityFormFields = DeepReplace<
+    ActivityResponseBody,
+    RawActivityItem,
+    ActivityItem
+>;
+
+export type PartialActivityItem = PartialForm<ActivityItem, 'client_id'>;
+type A = PartialActivityItem['custom_action'];
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's remove unused types

File: src/views/ThreeWActivityForm/schema.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            start_date: { required: true },
+            status: {},
+        };
+
+        schema = addCondition(
+            schema,
+            value,
+            ['start_date', 'end_date'] as const,
+            ['end_date'] as const,
+            (props) => {
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's define return type

File: src/views/ThreeWActivityForm/schema.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            schema,
+            value,
+            ['activity_lead'] as const,
+            [
+                'reporting_ns',
+                'reporting_ns_contact_name',
+                'reporting_ns_contact_email',
+                'reporting_ns_contact_role',
+            ] as const,
+            (props) => {
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's define return type

File: src/views/ThreeWActivityForm/schema.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                };
+            },
+        );
+
+        schema = addCondition(
+            schema,
+            value,
+            ['sectors'] as const,
+            ['activities'] as const,
+            (props) => {
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's define return type

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,774 @@
...
+        ].filter(isDefined),
+    );
+
+    const disabled_total = sum(
+        [
+            disabled_total_male,
+            disabled_total_female,
+            disabled_total_other,
+        ].filter(isDefined),
+    );
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx]: Let's just use safeSum

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,774 @@
...
+                        Unknown
+                    </div>
+                    <div className={styles.header}>
+                        Total
+                    </div>
+                </div>
+                <div className={styles.row}>
+                    <div className={_cs(styles.cell, styles.header)}>
+                        Male
+                    </div>
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx]: Let's use translation

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,774 @@
...
+                    <NumberInput
+                        className={styles.cell}
+                        inputSectionClassName={styles.inputSection}
+                        name="male_0_1_count"
+                        onChange={setFieldValue}
+                        value={value?.male_0_1_count}
+                        error={error?.male_0_1_count}
+                    />
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx:276]: Let's pass disabled one each input

File: src/views/ThreeWActivityForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+            (sector) => sector.id,
+            (sector) => sector,
+        )
+    ), [optionsResponse?.sectors]);
+
+    return (
+        <div className={styles.threeWActivityForm}>
+            <InputSection
+                title="IFRC supported Operation"
+                description="If operation does not appear in the dropdown, the operation does not yet exist in GO. In that case, please submit a new Field Report to generate the operation, then come back to this form"
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's use translation or at least define TODO/FIXME note

File: src/views/ThreeWActivityForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+                        name="deployed_eru"
+                        value={value?.deployed_eru}
+                        onChange={setFieldValue}
+                        options={erusResponse?.results}
+                        listContainerClassName={styles.radio}
+                        keySelector={(d) => d.id}
+                        labelSelector={(d) => (
+                            `${d.eru_owner?.national_society_country?.society_name}
+                            (${d.type_display})`
+                        )}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's not use inline functions

File: src/views/ThreeWActivityForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+                    <RadioInput
+                        name="deployed_eru"
+                        value={value?.deployed_eru}
+                        onChange={setFieldValue}
+                        options={erusResponse?.results}
+                        listContainerClassName={styles.radio}
+                        keySelector={(d) => d.id}
+                        labelSelector={(d) => (
+                            `${d.eru_owner?.national_society_country?.society_name}
+                            (${d.type_display})`
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: We should avoid string templates with value that can be undefined

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+        error: errorFromProps,
+        value,
+        type,
+    } = props;
+
+    const setFieldValue = useFormObject(mainIndex, onChange, {
+        client_id: randomString(),
+        action: actionDetails?.id,
+        sector: sectorKey,
+    });
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: THis is not correct. The object should be a function.

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+    }, [onChange, mainIndex]);
+
+    const pointCountInputShown = (
+        (value?.is_simplified_report && type === 'custom')
+        || (
+            value?.is_simplified_report
+            && !actionDetails?.is_cash_type
+            && actionDetails?.has_location
+        )
+    );
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: Why check for the same value is_simplified_report twice?

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+                <Button
+                    name={clientId}
+                    onClick={handleRemoveClick}
+                    variant="secondary"
+                    icons={(
+                        <DeleteBinLineIcon />
+                    )}
+                >
+                    Delete
+                </Button>
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: Let's use translation

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx (hunk 12 above)
Comment (AdityaKhatri) [reply src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: To clearly distinguish what happens for custom or action type

File: src/components/Checkbox/index.tsx [hunk 14]
```diff
@@ -88,14 +91,15 @@ function Checkbox<const NAME>(props: Props<NAME>) {
...
-            )}
-            {error && (
-                <div className={labelContainerClassName}>
-                    {error}
+                    {/* FIXME: Add better structure for error @frozenhelium */}
+                    {error && (
+                        <InputError className={_cs(styles.inputError, errorContainerClassName)}>
+                            {error}
+                        </InputError>
+                    )}
```
Comment (tnagorra) [inline src/components/Checkbox/index.tsx]: Any reason we changed the datastructure here?

File: src/views/ThreeWActivityForm/styles.module.css [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+        .sector-checkboxes {
+            display: flex;
+            flex-wrap: wrap;
+            gap: var(--go-ui-spacing-sm);
+            min-height: unset;
+
+            >* {
+                flex-basis: calc(100% / 3 - 2 * var(--go-ui-spacing-sm) / 3);
+            }
+        }
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/styles.module.css:26]: @frozenhelium Should we instead use grids?

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx [hunk 16]
```diff
@@ -0,0 +1,82 @@
+import {
+    ArrayError,
+    useFormObject,
+    SetValueArg,
+    getErrorObject,
+} from '@togglecorp/toggle-form';
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx]: When importing types let's indicate that they are types
```suggestion
import {
    type ArrayError,
    useFormObject,
    type SetValueArg,
    getErrorObject,
} from '@togglecorp/toggle-form';
```

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+import Button from '#components/Button';
+import NumberInput from '#components/NumberInput';
+import SelectInput from '#components/SelectInput';
+import { PartialActionSupplyItem } from '../../../schema';
+
+import styles from './styles.module.css';
+
+const keySelector = (item: { id: number }) => String(item.id);
+const labelSelector = (item: { title: string }) => item.title;
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx:19]: Let's use selectors @frozenhelium ?

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    const setFieldValue = useFormObject(index, onChange, () => ({
+        client_id: randomString(),
+    }));
+
+    const error = (value && value.client_id && errorFromProps)
+        ? getErrorObject(errorFromProps[value.client_id])
+        : undefined;
+
+    return (
+        <div className={styles.actionSupplyInput}>
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx:54]: Let's display non field errors.

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+                value={value?.supply_value}
+                error={error?.supply_value}
+                onChange={setFieldValue}
+            />
+            <Button
+                name={index}
+                variant="tertiary"
+                onClick={onRemove}
+            >
+                <DeleteBinLineIcon />
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx]: Let's add label on IconButton

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/CustomSuppliesInput/index.tsx [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+    const setFieldValue = useFormObject(index, onChange, () => ({
+        client_id: randomString(),
+    }));
+    const error = (value && value.client_id && errorFromProps)
+        ? getErrorObject(errorFromProps[value.client_id])
+        : undefined;
+
+    return (
+        <div className={styles.customSupplyInput}>
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/CustomSuppliesInput/index.tsx:45]: Let's display non field errors

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx [hunk 21]
```diff
@@ -0,0 +1,755 @@
...
+    PartialActivityItem,
+} from '../../../schema';
+
+import styles from './styles.module.css';
+
+interface Props {
+    value: PartialActivityItem;
+    setFieldValue: (...entries: EntriesAsList<PartialActivityItem>) => void;
+    error?: ObjectError<PartialActivityItem>;
+}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/DisaggregationInput/index.tsx]: Let's pass disabled prop on each input

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 22]
```diff
@@ -0,0 +1,525 @@
...
+        (
+            value?.is_simplified_report
+            && type === 'custom'
+        ) || (
+            value?.is_simplified_report
+            && type === 'action'
+            && !actionDetails?.is_cash_type
+            && actionDetails?.has_location
+        )
+    );
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: Let's simplify this boolean expression

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 23]
```diff
@@ -0,0 +1,525 @@
...
+        (
+            !value?.is_simplified_report
+            && type === 'custom'
+        ) || (
+            !value?.is_simplified_report
+            && type === 'action'
+            && !actionDetails?.is_cash_type
+            && actionDetails?.has_location
+        )
+    );
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: Let's simplify this boolean expression

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 24]
```diff
@@ -0,0 +1,525 @@
...
+    const budgetInputsShown = (
+        type === 'action'
+        && actionDetails?.is_cash_type
+    );
+
+    // FIXME: Write this condition appropriately
+    // This should be shown only if data is not filled
+    // by the user
+    const showNoDataAvailable = error?.has_no_data_on_people_reached
+    || value?.has_no_data_on_people_reached;
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]: Let's implement this validation using form's new context feature.
https://github.com/toggle-corp/toggle-form#useform

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 25]
```diff
@@ -0,0 +1,525 @@
...
+                    title="Delete activity"
+                    icons={(
+                        <DeleteBinLineIcon />
+                    )}
+                >
+                    {/* FIXME: Add translations */}
+                    Delete
+                </Button>
+            )}
+            heading={type === 'custom' ? `Custom Activity #${itemNumber}` : actionDetails?.title}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx:277]: Let's add "FIXME" for missing translation

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 26]
```diff
@@ -0,0 +1,525 @@
...
+                        <NumberInput
+                            name="people_count"
+                            // FIXME: Add translations
+                            label="People"
+                            value={value?.people_count}
+                            onChange={setFieldValue}
+                            error={error?.people_count}
+                            disabled={peopleCountDisabled}
+                        />
+                        OR
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx:341]: Let's add FIXME for translation

File: src/views/ThreeWActivityForm/index.tsx [hunk 27]
```diff
@@ -0,0 +1,799 @@
...
+
+    return 'on_going';
+}
+const activityLeadKeySelector = (item: NonNullable<GlobalEnums['deployments_emergency_project_activity_lead']>[number]) => item.key;
+const valueSelector = (item: { value: string }) => item.value;
+const idSelector = (item: { id: number }) => item.id;
+const deployedEruLabelSelector = (item: NonNullable<EruResponse['results']>[number]) => (
+    `${item.eru_owner?.national_society_country?.society_name}
+    (${item.type_display})`
+);
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's use selectors @frozenhelium ?

File: src/views/ThreeWActivityForm/index.tsx [hunk 28]
```diff
@@ -0,0 +1,799 @@
...
+        pathVariables: isTruthyString(activityId) ? {
+            id: Number(activityId),
+        } : undefined,
+        onSuccess: (response) => {
+            setDistrictOptions(response.districts_details);
+            setEventOptions([{
+                ...response.event_details,
+                // FIXME: event dtype id is a must inside event mini but
+                // its not defined under event_detail of this response
+                dtype: { id: response.event_details?.dtype } as EventItem['dtype'],
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: What is the further action?

File: src/views/ThreeWActivityForm/index.tsx [hunk 29]
```diff
@@ -0,0 +1,799 @@
...
+                            value={value?.reporting_ns}
+                            error={error?.reporting_ns}
+                        />
+                    </InputSection>
+                    <InputSection
+                        // FIXME: Add translation
+                        title="Contact Information"
+                        // FIXME: Add translation
+                        description="Who should be contacted for
+                        any coordination matters related to this response activity?"
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's join this in same line

File: src/views/ThreeWActivityForm/index.tsx [hunk 30]
```diff
@@ -0,0 +1,799 @@
...
+                            <Button
+                                name={undefined}
+                                onClick={handleFinalSubmitClick}
+                                disabled={createProjectPending || updateProjectPending}
+                            >
+                                {/* FIXME: Use translations */}
+                                Submit
+                            </Button>
+                            <div className={styles.note}>
+                                If you have any questions, contact the IM team &nbsp;
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx:770]: Let's add a fixme for translation

File: src/views/ThreeWActivityForm/index.tsx [hunk 31]
```diff
@@ -0,0 +1,799 @@
...
+                            ) : (
+                                selectedNationalSocietyDetail?.society_name
+                            )}
+                        />
+                        <TextOutput
+                            className={styles.metaItem}
+                            labelClassName={styles.metaLabel}
+                            valueClassName={styles.sectorsList}
+                            label="Actions Taken"
+                            value={value?.sectors?.map((sectorId) => (
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's add FIXME for translations

File: src/views/ThreeWActivityForm/index.tsx [hunk 32]
```diff
@@ -0,0 +1,799 @@
...
+                                    {(
+                                        value?.activities
+                                            ?.filter((activity) => (
+                                                activity.sector === sectorId
+                                                && isDefined(activity.action)
+                                            ))
+                                            .map((activity) => (
+                                                <TextOutput
+                                                    icon={(<LegendIcon />)}
+                                                    value={(
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Missing key prop in list

File: src/views/ThreeWActivityForm/schema.ts [hunk 33]
```diff
@@ -0,0 +1,1279 @@
...
+import type { paths } from '#generated/types';
+
+function countGreaterThanLength(count: number) {
+    return (value: Maybe<number[]>) => {
+        if (count < (value?.length ?? 0)) {
+            return 'There should be atleast one activity in selected sectors.';
+        }
+        return undefined;
+    };
+}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Why not use a generic greaterThanCondition?

cc: @frozenhelium

File: src/views/ThreeWActivityForm/schema.ts [hunk 34]
```diff
@@ -0,0 +1,1279 @@
...
+
+function hasValue(x: number) {
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's change the name to hasValueCondition

File: src/views/ThreeWActivityForm/schema.ts [hunk 35]
```diff
@@ -0,0 +1,1279 @@
...
+    supply_value: number;
+}
+
+type ActionSupplyItem = {
+    client_id: string;
+    supply_action: string;
+    supply_value: number;
+}
+
+export type ActivityResponseBody = paths['/api/v2/emergency-project/{id}/']['put']['requestBody']['content']['application/json'];
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's use type GoApiBody from '#utils/restRequest'

File: src/views/ThreeWActivityForm/schema.ts [hunk 36]
```diff
@@ -0,0 +1,1279 @@
...
+            event: { required: true },
+            activity_lead: { required: true },
+            country: { required: true },
+            districts: { defaultValue: [] },
+            start_date: { required: true },
+            status: {},
+            sectors: {
+                forceValue: undefinedValue,
+                validations: [countGreaterThanLength(sectorsInActivities?.length ?? 0)],
+            },
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: This should be written using addCondition as the sectors validation is dependent on activities key

File: src/components/Checkbox/index.tsx (hunk 14 above)
Comment (frozenhelium) [inline src/components/Checkbox/index.tsx]: We should move error out of label

File: src/views/ThreeWActivityForm/schema.ts [hunk 37]
```diff
@@ -0,0 +1,1279 @@
...
+                        deployed_eru: { required: true },
+                    };
+                }
+                return {
+                    reporting_ns: { forceValue: undefinedValue },
+                    deployed_eru: { forceValue: undefinedValue },
+                    reporting_ns_contact_name: { forceValue: undefinedValue },
+                    reporting_ns_contact_role: { forceValue: undefinedValue },
+                    reporting_ns_contact_email: { forceValue: undefinedValue },
+                };
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: A fool proof method to do this is first create "forceNull" fields
Then update them to have values on each branch..
Also why are we using `forceValue: undefinedValue`

File: src/views/ThreeWActivityForm/schema.ts [hunk 38]
```diff
@@ -0,0 +1,1279 @@
...
+                                    activityValue,
+                                    ['is_simplified_report'] as const,
+                                    ['points'] as const,
+                                    (val): Pick<ActivityItemSchemaFields, 'points'> => {
+                                        if (!val?.is_simplified_report) {
+                                            return {
+                                                points: {
+                                                    // FIXME: Have at least one point
+                                                    keySelector: (point) => (
+                                                        point.client_id as string
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Why do we need to cast this?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx [hunk 39]
```diff
@@ -0,0 +1,93 @@
...
+        : undefined;
+
+    return (
+        <div className={styles.actionSupplyInput}>
+            <NonFieldError error={error} />
+            <SelectInput
+                // FIXME: Add translation
+                label="Supply"
+                name="supply_action"
+                value={value?.supply_action ? String(value?.supply_action) : undefined}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/ActionSupplyInput/index.tsx]: Why cast to string?

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/index.tsx [hunk 40]
```diff
@@ -0,0 +1,224 @@
...
+                <Button
+                    name={undefined}
+                    onClick={handleCustomActivityAdd}
+                    variant="secondary"
+                    disabled={disabled}
+                    icons={(
+                        <AddLineIcon />
+                    )}
+                >
+                    Add custom activity
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/index.tsx:188]: Let's translate this

File: src/views/ThreeWProjectForm/index.tsx [hunk 41]
```diff
@@ -327,6 +327,14 @@ export function Component() {
         setFieldValue(true, 'is_annual_report');
     }, [setFieldValue]);
 
+    const handleProjectStatusChange = useCallback((newVal: boolean) => {
+        setValue((oldVal) => ({
+            ...oldVal,
+            status: newVal ? PROJECT_STATUS_COMPLETED : oldVal.status,
```
Comment (tnagorra) [inline src/views/ThreeWProjectForm/index.tsx]: Looks like the logic is flawed.

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 42]
```diff
@@ -0,0 +1,540 @@
...
+
+    const {
+        deployments_emergency_project_activity_people_households: peopleHouseholdOptions,
+    } = useGlobalEnums();
+
+    const error = getErrorObject(errorFromProps);
+    const peopleCountDisabled = isDefined(value?.male_count) || isDefined(value?.female_count);
+    const genderDisaggregationDisabled = isDefined(value?.people_count)
+    && isNotDefined(value?.male_count)
+    && isNotDefined(value?.female_count);
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]:
```suggestion
        && isNotDefined(value?.male_count)
        && isNotDefined(value?.female_count);
```

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 43]
```diff
@@ -0,0 +1,540 @@
...
+    );
+
+    const budgetInputsShown = (
+        type === 'action'
+        && actionDetails?.is_cash_type
+    );
+
+    // NOTE: This should be shown only if data is not filled by the user
+    const showNoDataAvailable = error?.has_no_data_on_people_reached
+    || value?.has_no_data_on_people_reached;
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]:
```suggestion
        || value?.has_no_data_on_people_reached;
```

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 44]
```diff
@@ -0,0 +1,540 @@
...
+                                title="Add locations"
+                                onClick={handleAddPointButtonClick}
+                            >
+                                {/* FIXME: Add translations */}
+                                Add Location
+                            </Button>
+
+                        )}
+                    >
+                        {value?.points?.map((p, i) => (
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]:
```suggestion
                        {value?.points?.map((point, i) => (
```

File: src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx [hunk 45]
```diff
@@ -0,0 +1,540 @@
...
+                                disabled={disabled}
+                                icons={(
+                                    <AddLineIcon />
+                                )}
+                                onClick={handleAddActionSupplyClick}
+                            >
+                                {/* FIXME: Add translations */}
+                                Add Action Supply
+                            </Button>
+
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/ActivitiesBySectorInput/ActivityInput/index.tsx]:
```suggestion
```

File: src/views/ThreeWActivityForm/index.tsx [hunk 46]
```diff
@@ -0,0 +1,839 @@
...
+        method: 'POST',
+        body: (ctx: ActivityResponseBody) => ctx,
+        onSuccess: (response) => {
+            alert.show(
+                // FIXME: Add translations
+                'Successfully created a response activity.',
+                { variant: 'success' },
+            );
+            hideSubmitConfirmation();
+            setFinalValues(undefined);
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Why reset the final values when we are navigating outside the page

File: src/views/ThreeWActivityForm/index.tsx [hunk 47]
```diff
@@ -0,0 +1,839 @@
...
+            id: Number(activityId),
+        },
+        onSuccess: () => {
+            alert.show(
+                // FIXME: Add translations
+                'Successfully updated activities',
+                { variant: 'success' },
+            );
+            hideSubmitConfirmation();
+            setFinalValues(undefined);
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's navigate to new page

File: src/views/ThreeWActivityForm/index.tsx [hunk 48]
```diff
@@ -0,0 +1,839 @@
...
+        },
+        onFailure: ({
+            value: { messageForNotification },
+            debugMessage,
+        }) => {
+            // FIXME: Add appropriate error handling
+            alert.show(
+                'Failed to update project activities',
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx:305]: Let's add translation

File: src/views/ThreeWActivityForm/index.tsx [hunk 49]
```diff
@@ -0,0 +1,839 @@
...
+            return {
+                ...oldVal,
+                end_date: newDate,
+                status,
+            };
+        });
+    }, [setValue]);
+
+    const handleSubmitClick = useCallback(() => {
+        beforeSubmitRef.current = false;
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx:406]: Let's add NOTE here

File: src/views/ThreeWActivityForm/index.tsx [hunk 50]
```diff
@@ -0,0 +1,839 @@
...
+                            activity?.custom_supplies,
+                            (item) => item.supply_label,
+                            (item) => item.supply_value,
+                        ),
+                        supplies: listToMap(
+                            activity?.supplies,
+                            (item) => item.supply_action,
+                            (item) => item.supply_value,
+                        ),
+                    })).filter((activity) => sectorValuesMap?.[activity.sector]),
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: We have to clear the activities values when sector is unticked. And we can remove this logic

File: src/views/ThreeWActivityForm/index.tsx [hunk 51]
```diff
@@ -0,0 +1,839 @@
...
+                        </span>
+                        {/* FIXME: Use translations */}
+                        &nbsp;emergency. Please review your selections below before submission.
+                    </div>
+                    <div className={styles.meta}>
+                        <TextOutput
+                            className={styles.metaItem}
+                            labelClassName={styles.metaLabel}
+                            valueClassName={styles.metaValue}
+                            label="Country"
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx:798]: Let's translate this

File: src/views/ThreeWActivityForm/index.tsx [hunk 52]
```diff
@@ -0,0 +1,839 @@
...
+                            labelClassName={styles.metaLabel}
+                            valueClassName={styles.sectorsList}
+                            // FIXME: Use translations
+                            label="Actions Taken"
+                            value={value?.sectors?.map((sectorId) => (
+                                <div
+                                    className={styles.sector}
+                                    key={sectorId}
+                                >
+                                    {sectorOptionsMap?.[sectorId].title}
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]:
```suggestion
                                    {sectorOptionsMap?.[sectorId].title ?? '--'}
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/ThreeWActivityForm/schema.ts [hunk 53]
```diff
@@ -0,0 +1,1299 @@
...
+        schema = addCondition(
+            schema,
+            value,
+            ['activities', 'sectors'] as const,
+            ['sectors'] as const,
+            (val): Pick<FormSchemaFields, 'sectors'> => {
+                const sectorsInActivities = unique(
+                    val?.activities?.map((activity) => activity.sector).filter(isDefined) ?? [],
+                    (item) => item,
+                );
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts:140]: Need to check why this addCondition is not working

File: src/views/ThreeWActivityForm/schema.ts [hunk 54]
```diff
@@ -0,0 +1,1299 @@
...
+                                            return {
+                                                custom_action: { forceValue: nullValue },
+                                                beneficiaries_count: {},
+                                                amount: {
+                                                    validations: [positiveIntegerCondition],
+                                                },
+                                                // TODO: write validation for duplicate labels
+                                                supplies: {
+                                                    keySelector: (supply) => (
+                                                        supply.client_id as string
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Why do we need to cast this?

File: src/views/ThreeWActivityForm/schema.ts [hunk 55]
```diff
@@ -0,0 +1,1299 @@
...
+                                                female_18_59_count: {},
+                                                female_60_plus_count: {},
+                                                female_unknown_age_count: {},
+                                                other_0_1_count: {},
+                                                other_2_5_count: {},
+                                                other_6_12_count: {},
+                                                other_13_17_count: {},
+                                                other_18_59_count: {},
+                                                other_60_plus_count: {},
+                                                other_unknown_age_count: {},
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: Let's add positiveIntegerCondition for all count fields

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/ThreeWActivityForm/schema.ts [hunk 56]
```diff
@@ -134,6 +134,7 @@ const finalSchema: FormSchema = {
             ['activities', 'sectors'] as const,
             ['sectors'] as const,
             (val): Pick<FormSchemaFields, 'sectors'> => {
+                // FIXME: This condition is not triggered after value change
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/schema.ts]: This has been fixed. Let's remove the FIXME.
The issue was caused by improper usage of setValue function

File: src/views/ThreeWProjectForm/index.tsx [hunk 57]
```diff
@@ -77,6 +77,33 @@ import AnnualSplitInput from './AnnualSplitInput';
 import styles from './styles.module.css';
 import i18n from './i18n.json';
 
+type ProjectStatus = NonNullable<GlobalEnums['deployments_project_status']>[number];
+
+function calculateStatus(
```
Comment (tnagorra) [inline src/views/ThreeWProjectForm/index.tsx:82]: Is this function re-used?

File: src/views/ThreeWActivityForm/index.tsx [hunk 58]
```diff
@@ -371,18 +376,39 @@ export function Component() {
...
+                () => true,
+            );
+
+            return {
+                ...oldVal,
+                sectors: newSectors,
+                activities: oldVal
+                    ?.activities
+                    ?.filter((activity) => activity.sector && sectorValuesMap?.[activity.sector]),
+            };
```
Comment (tnagorra) [inline src/views/ThreeWActivityForm/index.tsx]: Let's add a NOTE why we are doing this?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**